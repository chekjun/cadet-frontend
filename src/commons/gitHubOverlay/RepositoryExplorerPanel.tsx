import { Classes, Radio, RadioGroup } from '@blueprintjs/core';
import classNames from 'classnames';
import { useEffect } from 'react';

export const RepositoryExplorerPanel = (props: any) => {
  const { userRepos, repoName, setRepoName, setRepoFiles } = props;

  useEffect(() => {
    setRepoFiles();
  }, [repoName, setRepoFiles]);

  return (
    <div className={classNames(Classes.DIALOG_BODY, 'repo-step')}>
      <RadioGroup onChange={setRepoName} selectedValue={repoName}>
        {userRepos.map((repo: any) => (
          <Radio label={repo.name} key={repo.id} value={repo.name} />
        ))}
      </RadioGroup>
    </div>
  );
};
