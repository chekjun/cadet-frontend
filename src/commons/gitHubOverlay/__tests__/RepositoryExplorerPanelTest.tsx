import { mount } from 'enzyme';

import { RepositoryExplorerPanel } from '../RepositoryExplorerPanel';

const REPO_1 = { name: 'repoName1', id: 1 };
const REPO_2 = { name: 'repoName2', id: 2 };
const REPO_3 = { name: 'repoName3', id: 3 };

const USER_REPOS: { name: string; id: number }[] = [REPO_1, REPO_2, REPO_3];
const EMPTY_USER_REPOS: { name: string; id: number }[] = [];

test('Test repository list renders correctly', () => {
  const props = {
    userRepos: USER_REPOS,
    repoName: '',
    setRepoName: () => {},
    refreshRepoFiles: () => {}
  };
  const REP = mount(<RepositoryExplorerPanel {...props} />);
  expect(REP.debug()).toMatchSnapshot();
});

test('Test empty repository list renders correctly', () => {
  const props = {
    userRepos: EMPTY_USER_REPOS,
    repoName: '',
    setRepoName: () => {},
    refreshRepoFiles: () => {}
  };
  const REP = mount(<RepositoryExplorerPanel {...props} />);
  expect(REP.debug()).toMatchSnapshot();
});
