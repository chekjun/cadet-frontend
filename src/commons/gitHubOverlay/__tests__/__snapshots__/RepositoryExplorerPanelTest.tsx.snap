// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test empty repository list renders correctly 1`] = `
"<RepositoryExplorerPanel userRepos={{...}} repoName=\\"\\" setRepoName={[Function: setRepoName]} refreshRepoFiles={[Function: refreshRepoFiles]}>
  <div className=\\"bp3-dialog-body repo-step\\">
    <Blueprint3.RadioGroup onChange={[Function: setRepoName]} selectedValue=\\"\\">
      <div className={[undefined]} />
    </Blueprint3.RadioGroup>
  </div>
</RepositoryExplorerPanel>"
`;

exports[`Test repository list renders correctly 1`] = `
"<RepositoryExplorerPanel userRepos={{...}} repoName=\\"\\" setRepoName={[Function: setRepoName]} refreshRepoFiles={[Function: refreshRepoFiles]}>
  <div className=\\"bp3-dialog-body repo-step\\">
    <Blueprint3.RadioGroup onChange={[Function: setRepoName]} selectedValue=\\"\\">
      <div className={[undefined]}>
        <Blueprint3.Radio label=\\"repoName1\\" value=\\"repoName1\\" checked={false} className={[undefined]} disabled={[undefined]} inline={[undefined]} name=\\"Blueprint3.RadioGroup-0\\" onChange={[Function: setRepoName]}>
          <Control label=\\"repoName1\\" value=\\"repoName1\\" checked={false} className={[undefined]} disabled={[undefined]} inline={[undefined]} name=\\"Blueprint3.RadioGroup-0\\" onChange={[Function: setRepoName]} type=\\"radio\\" typeClassName=\\"bp3-radio\\">
            <label className=\\"bp3-control bp3-radio\\" style={[undefined]}>
              <input value=\\"repoName1\\" checked={false} disabled={[undefined]} name=\\"Blueprint3.RadioGroup-0\\" onChange={[Function: setRepoName]} type=\\"radio\\" />
              <span className=\\"bp3-control-indicator\\" />
              repoName1
            </label>
          </Control>
        </Blueprint3.Radio>
        <Blueprint3.Radio label=\\"repoName2\\" value=\\"repoName2\\" checked={false} className={[undefined]} disabled={[undefined]} inline={[undefined]} name=\\"Blueprint3.RadioGroup-0\\" onChange={[Function: setRepoName]}>
          <Control label=\\"repoName2\\" value=\\"repoName2\\" checked={false} className={[undefined]} disabled={[undefined]} inline={[undefined]} name=\\"Blueprint3.RadioGroup-0\\" onChange={[Function: setRepoName]} type=\\"radio\\" typeClassName=\\"bp3-radio\\">
            <label className=\\"bp3-control bp3-radio\\" style={[undefined]}>
              <input value=\\"repoName2\\" checked={false} disabled={[undefined]} name=\\"Blueprint3.RadioGroup-0\\" onChange={[Function: setRepoName]} type=\\"radio\\" />
              <span className=\\"bp3-control-indicator\\" />
              repoName2
            </label>
          </Control>
        </Blueprint3.Radio>
        <Blueprint3.Radio label=\\"repoName3\\" value=\\"repoName3\\" checked={false} className={[undefined]} disabled={[undefined]} inline={[undefined]} name=\\"Blueprint3.RadioGroup-0\\" onChange={[Function: setRepoName]}>
          <Control label=\\"repoName3\\" value=\\"repoName3\\" checked={false} className={[undefined]} disabled={[undefined]} inline={[undefined]} name=\\"Blueprint3.RadioGroup-0\\" onChange={[Function: setRepoName]} type=\\"radio\\" typeClassName=\\"bp3-radio\\">
            <label className=\\"bp3-control bp3-radio\\" style={[undefined]}>
              <input value=\\"repoName3\\" checked={false} disabled={[undefined]} name=\\"Blueprint3.RadioGroup-0\\" onChange={[Function: setRepoName]} type=\\"radio\\" />
              <span className=\\"bp3-control-indicator\\" />
              repoName3
            </label>
          </Control>
        </Blueprint3.Radio>
      </div>
    </Blueprint3.RadioGroup>
  </div>
</RepositoryExplorerPanel>"
`;
